src/devices/machine/68153bim.h

```diff
@@ -0,0 +1,225 @@
+#define MCFG_BIM68153_OUT_IACK2_CB(_devcb)                              \
+	devcb = &bim68153::set_out_iack3_callback(*device, DEVCB_##_devcb);
+#define MCFG_BIM68153_OUT_IACK3_CB(_devcb)                              \
+	devcb = &bim68153::set_out_iack3_callback(*device, DEVCB_##_devcb);
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+class bim68153_device;
+
+class bim68153_channel : public device_t
+{
+	friend class bim68153_device;
+
+public:
+	bim68153_channel(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// device-level overrides
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	// read register handlers
+	uint8_t do_bimreg_control_r();
+	uint8_t do_bimreg_vector_r();
+
+	// write register handlers
+	void do_bimreg_control_w(uint8_t data);
+	void do_bimreg_vector_w(uint8_t vector);
+
+	DECLARE_WRITE_LINE_MEMBER( int_w );
+	devcb_read8 m_out_iack_cb;
+	uint8_t m_int_state;
+
+	// Register state
+	uint8_t m_control;
+	uint8_t m_vector;
+
+protected:
+	enum
+	{
+		NONE     = 0,
+		PENDING  = 1,
+	};
+
+	enum
+	{
+		RCV_IDLE     = 0,
+		RCV_SEEKING  = 1,
+		RCV_SAMPLING = 2
+	};
+
+	enum
+	{
+		REG_CNTRL_INT_LVL_MSK   = 0x07,
+		REG_CNTRL_INT_AUT_DIS   = 0x08,
+		REG_CNTRL_INT_ENABLE    = 0x10,
+		REG_CNTRL_INT_EXT       = 0x20,
+		REG_CNTRL_INT_AUT_CLR   = 0x40,
+		REG_CNTRL_INT_FLAG      = 0x80,
+	};
+	int m_int;      // interrupt request from connected device
+	int m_index;    // Which channel am I?
+	bim68153_device *m_bim; // my device
+};
+
+
+class bim68153_device :  public device_t
+{
+	friend class bim68153_channel;
+
+public:
+	// construction/destruction
+	bim68153_device(const machine_config &mconfig, device_type type, const char *name, const char *tag, device_t *owner, uint32_t clock, uint32_t variant, const char *shortname, const char *source);
+	bim68153_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	IRQ_CALLBACK_MEMBER(iack);
+	int acknowledge();
+	int get_irq_level();
+
+	template<class _Object> static devcb_base &set_out_int_callback(device_t &device, _Object object) { return downcast<bim68153_device &>(device).m_out_int_cb.set_callback(object); }
+	template<class _Object> static devcb_base &set_out_intal0_callback(device_t &device, _Object object) { return downcast<bim68153_device &>(device).m_out_intal0_cb.set_callback(object); }
+	template<class _Object> static devcb_base &set_out_intal1_callback(device_t &device, _Object object) { return downcast<bim68153_device &>(device).m_out_intal1_cb.set_callback(object); }
+
+	template<class _Object> static devcb_base &set_out_iack0_callback(device_t &device, _Object object) { return downcast<bim68153_device &>(device).m_chn[CHN_0]->m_out_iack_cb.set_callback(object); }
```

Comment:
The segfaults are because required_device finders can't be accessed before device_start (except by the validity checker). Try using subdevice instead.